src/output/file_name.rs

```diff
@@ -351,13 +351,22 @@ impl<'a, 'dir, C: Colours> FileName<'a, 'dir, C> {
         let mut display_hyperlink = false;
         if self.options.embed_hyperlinks == EmbedHyperlinks::On {
             if let Some(abs_path) = self.file.absolute_path.as_ref().and_then(|p| p.as_os_str().to_str()) {
+                #[cfg(not(target_os = "windows"))]
                 bits.insert(0, ANSIString::from(format!(
                     "{}file://{}{}{}",
                     HYPERLINK_START,
                     gethostname::gethostname().to_str().unwrap_or(""),
                     urlencoding::encode(abs_path).replace("%2F", "/"),
                     HYPERLINK_END,
                 )));
+                #[cfg(target_os = "windows")]
+                bits.insert(0, ANSIString::from(format!(
+                    "{}file://{}{}",
+                    HYPERLINK_START,
+                    abs_path.replace("\\\\?\\", ""),
```

Comment:
What does the `.replace()` part do? It’s unclear to me. @daviessm

Reply:
`Path::canonicalize` is an alias for `fs::canonicalize` and on Windows that returns a path that begins with `\\?\`, e.g. `\\?\C:\Users`. I couldn't find any documentation on how or whether that works with `file://` URLs and from my testing it doesn't, so I just stripped it off. This means that very long paths will probably not work, but they break things in Windows anyway so it isn't really a big issue.